### PR TITLE
Add nightly job

### PR DESCRIPTION
**Pull request by TylerJang27**
This will run against all existing snapshots AND against the latest version of each linter.

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        linter-version: [Snapshots, Latest]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+        # TODO(Tyler): Determine if we want to wipe the download cache and start fresh each night.
```
Comment (TylerJang27) [inline .github/workflows/nightly.yaml:20]: Do we want to wipe out the contents of the cache? Is the easiest way to do that just rm?

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: Nightly
+on:
+  schedule:
```
Comment (sxlijin) [inline .github/workflows/nightly.yaml:3]: nice-to-have: also run this workflow on changes to this file (we use paths-filter for this in the monorepo's `pr.yaml`)

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run plugin tests
+        run: |
+          PATH=$(npm bin):$PATH PLUGINS_TEST_LINTER_VERSION=${{ matrix.linter-version }} npm test --ci
```
Comment (sxlijin) [inline .github/workflows/nightly.yaml:39]: i think using the native mechanism where possible is cleaner here:

```suggestion
      - name: Run plugin tests
        run: |
          PATH=$(npm bin):$PATH npm test --ci
        env:
          PLUGINS_TEST_LINTER_VERSION: ${{ matrix.linter-version }}
```

File: .github/workflows/nightly.yaml (hunk 1 above)
Comment (sxlijin) [reply .github/workflows/nightly.yaml:20]: Pretty much, yeah. Look at .github/actions/linux_build_and_test/action.yaml in the monorepo for an exaple.

File: .github/workflows/nightly.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+        # TODO(Tyler): Determine if we want to wipe the download cache and start fresh each night.
+
+      - name: Cache tool downloads
+        uses: actions/cache@v3
+        with:
+          path: /tmp/plugins_testing_download_cache
+          # No need to key on trunk version unless we change how we store downloads.
```
Comment (sxlijin) [inline .github/workflows/nightly.yaml:26]:
```suggestion
          # No need to key on trunk version or trunk.yaml unless we change how we store downloads.
```

in comparison to https://github.com/trunk-io/trunk-action/blob/2cda8a0c9923423307c2c3efe676a4c4614e46e6/action.yaml#L88-L91

**Review by sxlijin — APPROVED** [review]

File: .github/workflows/nightly.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+name: Nightly
+on:
+  schedule:
+    # Times in UTC (PST+8), this translates to running Mon-Fri at midnight/1am (depending on DST)
+    - cron: 0 8 * * 1-5
+  pull_request: {}
```
Comment (sxlijin) [inline .github/workflows/nightly.yaml:6]:
```suggestion
  pull_request:
    paths: [.github/workflows/nightly.yaml]
```

looked it up after, looks like you can apply the filter on the trigger itself

**Pull request merged**